src/main/java/roomescape/reservation/service/ReservationDeleteService.java

```diff
@@ -15,34 +19,57 @@ public class ReservationDeleteService {
 
     private final ReservationRepository reservationRepository;
     private final WaitingRepository waitingRepository;
+    private final PaymentRepository paymentRepository;
+    private final PaymentClient paymentClient;
 
-    public ReservationDeleteService(ReservationRepository reservationRepository, WaitingRepository waitingRepository) {
+    public ReservationDeleteService(ReservationRepository reservationRepository,
+                                    WaitingRepository waitingRepository,
+                                    PaymentRepository paymentRepository,
+                                    PaymentClient paymentClient) {
         this.reservationRepository = reservationRepository;
         this.waitingRepository = waitingRepository;
+        this.paymentRepository = paymentRepository;
+        this.paymentClient = paymentClient;
     }
 
     @Transactional
     public void deleteReservation(Long reservationId) {
         Reservation reservation = findReservation(reservationId);
         validateIsAfterFromNow(reservation);
 
+        refund(reservation);
         findHighPriorityWaiting(reservationId).ifPresentOrElse(
-                Waiting::confirmReservation,
-                () -> reservationRepository.deleteById(reservationId));
+                waiting -> confirmReservation(reservation, waiting),
+                () -> reservationRepository.delete(reservation));
     }
 
     private Reservation findReservation(Long id) {
         return reservationRepository.findById(id)
                 .orElseThrow(() -> new BadArgumentRequestException("해당 예약을 찾을 수 없습니다."));
     }
 
-    private Optional<Waiting> findHighPriorityWaiting(Long reservationId) {
-        return waitingRepository.findTopByReservationIdOrderByCreatedAtAsc(reservationId);
-    }
-
     private void validateIsAfterFromNow(Reservation reservation) {
         if (reservation.isBefore(LocalDateTime.now())) {
             throw new BadArgumentRequestException("예약은 현재 날짜 이후여야 합니다.");
         }
     }
+
+    private void refund(Reservation reservation) {
+        if (reservation.canRefund()) {
+            Payment payment = paymentRepository.findByScheduleAndMemberAndStatus(
+                            reservation.getSchedule(), reservation.getMember(), PaymentStatus.PAID)
+                    .orElseThrow(() -> new BadArgumentRequestException("해당 예약 내역이 존재하지 않습니다."));
+            paymentClient.refundPayment(payment.getPaymentKey());
+            payment.completeRefund();
+        }
+    }
+
+    private Optional<Waiting> findHighPriorityWaiting(Long reservationId) {
+        return waitingRepository.findTopByScheduleIdOrderByCreatedAtAsc(reservationId);
+    }
+
+    private void confirmReservation(Reservation reservation, Waiting waiting) {
+        reservation.reconfirmReservation(waiting.getMember());
+        waitingRepository.delete(waiting);
+    }
```

Comment:
```suggestion
    private void confirmReservation(Reservation reservation, Waiting waiting) {
        reservation.confirm(waiting.getMember());
        waitingRepository.delete(waiting);
    }
```

reseravation 입장에서는 구체적인 비즈니스 로직을 알필요가 없어보입니다.
멤버를 변경해주는 정도만 알고 있어도 괜찮지 않을까요?